app/service/planning_svc.py

```diff
@@ -210,3 +219,15 @@ def _reward_fact_relationship(combo_set, combo_link, score):
 
     async def _default_link_status(self, operation):
         return self.LinkState.EXECUTE.value if operation['autonomous'] else self.LinkState.PAUSE.value
+
+    async def _enforce_relationship(self, fact_combo, relationship):
+        for i in range(len(fact_combo) - 1):
+            if relationship.get('property1') == fact_combo[i]['property'] \
```

Comment:
may be more clear to use source/target here instead of prop1/prop2

Reply:
yeah I think that is prob better as well